Prevent erroneous using directives from blocking the initial run of BSP

Gedochao opened the pull request:
Fixes #1097

lwronski commented on modules/build/src/main/scala/scala/build/bsp/BspImpl.scala:
```diff
@@ -44,9 +62,19 @@ final class BspImpl(
+    *   options which may be reloaded on a bsp workspace/reload request
+    * @param isBspInitRun
+    *   a flag indicating if this is the BSP initial setup workflow
+    */
```
The name of parameter `isBspInitRun` and description should be more descriptive. 

To understand this flag, I first need to know how BSP initial setup workflow works

lwronski commented on modules/options/src/main/scala/scala/build/Artifacts.scala:
```diff
@@ -375,12 +380,19 @@ object Artifacts {
         .map(dep => dep.toCs(paramsOpt).map(csDep => (dep.value, csDep.value)))
+        .filter {
+          case Left(l) if isBspInitRun =>
+            logger.error(l.message)
```
```suggestion
          case Left(e) if isBspInitRun =>
            logger.error(e.message)
```

lwronski commented on modules/integration/src/test/scala/scala/cli/integration/BspTestDefinitions.scala:
```diff
@@ -1118,9 +1120,50 @@ abstract class BspTestDefinitions(val scalaVersionOpt: Option[String])
+              bloopProjectNames.foreach { bloopProjectName =>
+                val bloopProjectJsonPath = bloopDir / s"$bloopProjectName.json"
+                expect(os.isFile(bloopProjectJsonPath))
+              }
```
We should also test, if `bsp` properly generates error diagnostic.

Gedochao replied:
good call, testing for diagnostics made me realise those were broken on the initial run (recovering from errors prevented the silenced errors to be transferred as diagnostics).
ought to be fixed now.

Gedochao committed:
NIT Refactor
  - remove unnecessary brackets
  - update some syntax to Scala 3
  - add in some missing type annotations
  - other minor fixes
  - refactor tests for diagnostics in BSP

alexarchambault reviewed (commented):
NIT About the commit structure, so that we have a clean(ish) history on the main branch: could you create a separate commit for the added scaladoc, and squash the other non-refacto commits?

Gedochao committed:
Prevent erroneous using directives from blocking the initial run of BSP

Gedochao committed:
Add some scaladocs in BSP- and build-related classes

Gedochao commented:
> NIT About the commit structure, so that we have a clean(ish) history on the main branch: could you create a separate commit for the added scaladoc, and squash the other non-refacto commits?

@alexarchambault should be fixed now.
